File: python/7 kyu/362.py
# https://www.codewars.com/kata/558dd9a1b3f79dc88e000001/train/python

# Passed

def find_dup(arr):
    seen = set()
    result = None
    for num in arr:
        if num in seen:
            result = num
            break

        seen.add(num)
        
    return result

output = find_dup([8,2,6,3,7,2,5,1,4])
print(output)

File: nodejs/7 kyu/362.js
// https://www.codewars.com/kata/58b6c403a38abaaf6c00006b/train/javascript

// Passed

function sameEncryption(s1, s2) {
  let encryptString = (stringToEncrypt) => {
    let middleTerm = String(
      String(stringToEncrypt.slice(1, stringToEncrypt.length - 1).length),
    );
    while (middleTerm.length > 1) {
      middleTerm = middleTerm
        .split("")
        .map((digit) => Number(digit))
        .reduce((prev, curr) => prev + curr, 0);
      middleTerm = String(middleTerm);
    }
    let encryptedString =
      stringToEncrypt[0] +
      middleTerm +
      stringToEncrypt[stringToEncrypt.length - 1];
    return encryptedString;
  };

  let result = encryptString(s1) === encryptString(s2);
  return result;
}

const output = sameEncryption("fKhjuytrdfcdc", "flJc");
console.log(output);